src/aws_credentials_ec2.erl

```diff
@@ -3,40 +3,70 @@
 -module(aws_credentials_ec2).
 -behaviour(aws_credentials_provider).
 
+-define(SESSION_TOKEN_URL,
+        "http://169.254.169.254/latest/api/token/").
+-define(SESSION_TOKEN_TTL_HEADER,
+        "x-aws-ec2-metadata-token-ttl-seconds").
+-define(SESSION_TOKEN_TTL_SECONDS,
+        "21600").
+-define(SESSION_TOKEN_HEADER,
+        "X-aws-ec2-metadata-token").
 -define(CREDENTIAL_URL,
         "http://169.254.169.254/latest/meta-data/iam/security-credentials/").
 -define(DOCUMENT_URL,
         "http://169.254.169.254/latest/dynamic/instance-identity/document").
 
 -type role() :: binary().
+-type session_token() :: binary().
+% TODO: How can I get these types from httpc?
+-type field() :: string().
+-type value() :: string().
+-type request_header() :: {field(), value()}.
+-type request_headers() :: [request_header()].
 
 -export([fetch/1]).
 
 -spec fetch(aws_credentials_provider:options()) ->
         {ok, aws_credentials_provider:credentials(), aws_credentials_provider:expiration()} |
         {error, any()}.
 fetch(_Options) ->
-    {ok, Role} = fetch_role(),
-    {ok, AccessKeyID, SecretAccessKey, ExpirationTime, Token} = fetch_metadata(Role),
-    {ok, Region} = fetch_document(),
+    % Fetch the IMDS session token and convert it to the request headers.
```

Comment:
This is the main part of the PR.  Getting the token first and passing it along in a header on every request.